pkg/process/checks/pod.go

```diff
@@ -0,0 +1,113 @@
+// +build linux,kubelet
+
+package checks
+
+import (
+	"sync"
+	"time"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	yaml "gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Pod is a singleton PodCheck.
+var Pod = &PodCheck{}
+
+// PodCheck is a check that returns container metadata and stats.
+type PodCheck struct {
+	sync.Mutex
+
+	sysInfo                 *model.SystemInfo
+	containerFailedLogLimit *util.LogLimit
+}
+
+// Init initializes a PodCheck instance.
+func (c *PodCheck) Init(cfg *config.AgentConfig, info *model.SystemInfo) {
+	c.sysInfo = info
+	c.containerFailedLogLimit = util.NewLogLimit(10, time.Minute*10)
+}
+
+// Name returns the name of the ProcessCheck.
+func (c *PodCheck) Name() string { return "pod" }
+
+// Endpoint returns the endpoint where this check is submitted.
+func (c *PodCheck) Endpoint() string { return "/api/v1/orchestrator" }
+
+// RealTime indicates if this check only runs in real-time mode.
+func (c *PodCheck) RealTime() bool { return false }
+
+// Run runs the PodCheck to collect a list of running pods
+func (c *PodCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.MessageBody, error) {
+	c.Lock()
+	defer c.Unlock()
+
+	util, err := kubelet.GetKubeUtil()
+	if err != nil {
+		return nil, err
+	}
+
+	podList, err := util.GetRawLocalPodList()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("Collected %d pods", len(podList))
+
+	podMsgs := []*model.Pod{}
+
+	for p := 0; p < len(podList); p++ {
```

Comment:
a nit: any reason you're not using `for i := range podlist`?

Reply:
yes, using `range` would give me a copy of the values, here I want to modify the original object to benefit from the yaml marshalling on it.